frontend/src/containers/NotificationModal.vue

```diff
@@ -0,0 +1,171 @@
+  },
+  methods: {
+    setTab(tab) {
+      this.tab = tab;
+    },
+    changeCursor() {
+      this.$refs.allNotif.style.cursor = "pointer";
+    },
+    fetchNotifications() {
+        this.notifications = [];
+        axios({
+            methods: "GET",
+            url: `${this.$store.state.api}/events/dealership/${this.$store.state.auth.dealership}?eventType=${this.subscribedEvents}`,
+            headers: {
+            Authorization: `Bearer ${this.$store.state.auth.token}`,
+            },
+        })
+        .then((response) => {
+            if (response.data.success) {
+                response.data.payload.forEach((notif) => {
+                    var currentDateObj = new Date(notif.timestamp);
+                    notif.timestamp = currentDateObj.toString().split('GMT')[0];
+                    this.notifications.push(notif);
+                })
+            }
+        })
+        .catch((error) => {
+            console.log(error);
+        })
+    },
+    fetchUnReadNotif() {
+      // fetch new Unread events
+      this.unReadNotif= [];
+      axios({
+        methods: "GET",
+        url: `${this.$store.state.api}/events/dealership/${this.$store.state.auth.dealership}/user/${this.$store.state.auth.userId}`,
+        headers: {
+          Authorization: `Bearer ${this.$store.state.auth.token}`,
+        },
+      })
+      .then((response) => {
+        if (response.data.success) {
+          this.itemsCount = 0;
+          response.data.payload.forEach((notif) => {
+            var currentDateObj = new Date(notif.timestamp);
+            notif.timestamp = currentDateObj.toString().split('GMT')[0];
+            this.unReadNotif.push(notif);
+          })
+        }
+      })
+      .catch((error) => {
+        console.log(error);
+      })
+    },
+    redirect(notif) {
+      localStorage.setItem('vehicle', notif.vehicle)
+      var redirect = notif.title.indexOf('Vehicle') == -1 ? '/transactions' : '/inventory'
+      if (this.$route.path == '/transactions' || this.$route.path == '/inventory')
+        this.$router.go();
+      else
+        this.$router.push(redirect);
+    },
+    redirectAndMarkRead(notif, redirectOnly) {
+      if (!redirectOnly) {
+        axios({
+          method: "PUT",
+          url: `${this.$store.state.api}/events/${notif._id}/user/${this.$store.state.auth.userId}`,
+          headers: {
+            Authorization: `Bearer ${this.$store.state.auth.token}`,
+          },
+        })
+        .then((response) => {
+          if (response.data.success) {
+            console.log("success");
+          }
+        })
+        .catch((error) => {
+          console.log(error);
+        })
+        this.setNotifModal(false);
+        this.redirect(notif);
+      }
+      else {
+        this.redirect(notif);
+      } 
+    },
+  },
+  mounted() {
+      this.subscribedEvents = this.$store.state.auth.userEventsSubscriptions.join(',');
+      this.fetchNotifications();
+      this.fetchUnReadNotif();
```

Comment:
Is it necessary to do 2 different API calls for notifications? Would it be possible to do one call and filter the un read notifs from the list in the frontend?

Reply:
We already have an endpoint set up to return the unRead notifications, and one for all notification. I believe making the API call is faster in execution rather than filtering the list in the frontend.